force-app/main/default/classes/SOQL.cls

```diff
@@ -345,17 +524,28 @@ public with sharing class SOQL {
      */
     public static void setPrimaryContact() {
         //Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
-
+        List<Opportunity> opps = getOpportunityDetailsForClosedWon();
         // Create a set of Account Ids from the Opportunities
-
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity opp : opps) {
+            accountIds.add(opp.AccountId);
+        }
         // Query the Contacts
-
+        List<Contact> contacts = [SELECT AccountId, Id
+                                    FROM Contact
+                                    WHERE AccountId IN :accountIds];
+        // Create a map of Account Ids to Contacts
+        Map<Id, Id> accountToContactMap = new Map<Id, Id>();
         // Loop through each Contact and add the Account Id and Contact to the map
-        
+        for (Contact c : contacts) {
+            accountToContactMap.put(c.AccountId, c.Id);
+        }
         // Loop through each Opportunity
-
+            for (Opportunity opp : opps) {
                 // Update the Opportunity primary contact from accountToContactMap
-
+                opp.Primary_Contact__c = accountToContactMap.get(opp.AccountId);
         // Update the Opportunities
+                update opp;
```

Comment:
Here you need to add your opp to a list to be updated later. In other words, you need to bulkify your code here to avoid exceeding Salesforce governor limits. Bulkify this method.